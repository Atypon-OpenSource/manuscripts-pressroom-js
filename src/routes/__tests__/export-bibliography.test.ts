/*!
 * Â© 2020 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import request from 'supertest'

jest.mock('../../lib/jwt-authentication')

describe('export bibliography', () => {
  test('exports CSL JSON as BibTeX', async () => {
    const { app } = await import('../../app')

    const response = await request(app)
      .post('/api/v2/export/bibliography')
      .attach('file', __dirname + '/__fixtures__/csl.json')
      .field('format', 'bibtex')
      .responseType('blob')

    expect(response.status).toBe(200)
    expect(response.type).toBe('text/plain')
    expect(response.body.toString()).toMatchSnapshot('export-bibtex')
  })

  test('exports CSL JSON as RIS', async () => {
    const { app } = await import('../../app')

    const response = await request(app)
      .post('/api/v2/export/bibliography')
      .attach('file', __dirname + '/__fixtures__/csl.json')
      .field('format', 'ris')
      .responseType('blob')

    expect(response.status).toBe(200)
    expect(response.type).toBe('text/plain')
    expect(response.body.toString()).toMatchSnapshot('export-ris')
  })

  test('exports CSL JSON as EndNote', async () => {
    const { app } = await import('../../app')

    const response = await request(app)
      .post('/api/v2/export/bibliography')
      .attach('file', __dirname + '/__fixtures__/csl.json')
      .field('format', 'end')
      .responseType('blob')

    expect(response.status).toBe(200)
    expect(response.type).toBe('text/plain')
    expect(response.body.toString()).toMatchSnapshot('export-endnote')
  })
})
