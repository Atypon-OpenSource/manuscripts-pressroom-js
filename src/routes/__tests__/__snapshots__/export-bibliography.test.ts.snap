// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export bibliography exports CSL JSON as BibTeX: export-bibtex 1`] = `
"   @book{Piipari_Eaton_Pepe_2019, title={Introducing Ops Editor: open source editing environment for structured, computationally reproducible research documents}, url={https://www.ncbi.nlm.nih.gov/books/NBK540955/}, publisher={National Center for Biotechnology Information (US)}, author={Piipari, Matias and Eaton, Alf and Pepe, Alberto}, year={2019} }
"
`;

exports[`export bibliography exports CSL JSON as EndNote: export-endnote 1`] = `
"﻿%0 Book
%T Introducing Ops Editor: open source editing environment for structured, computationally reproducible research documents
%A Piipari, Matias
%A Eaton, Alf
%A Pepe, Alberto
%D 2019
%I National Center for Biotechnology Information (US)
%F Piipari_Eaton_Pepe_2019
%U https://www.ncbi.nlm.nih.gov/books/NBK540955/

"
`;

exports[`export bibliography exports CSL JSON as RIS: export-ris 1`] = `
"﻿TY  - BOOK
AU  - Piipari, Matias
AU  - Eaton, Alf
AU  - Pepe, Alberto
PY  - 2019
DA  - 2019//
TI  - Introducing Ops Editor: open source editing environment for structured, computationally reproducible research documents
PB  - National Center for Biotechnology Information (US)
UR  - https://www.ncbi.nlm.nih.gov/books/NBK540955/
ID  - Piipari_Eaton_Pepe_2019
ER  - 
"
`;
